appintro/src/main/java/com/github/paolorotolo/appintro/AppIntroBase.java

```diff
@@ -451,11 +482,48 @@ private void handleSlideChanged(Fragment oldFragment, Fragment newFragment) {
 
     /**
      * Called after a new slide has been selected
+     * {@link #canChangeLastSlide(int)} is called to check if the last slide can be changed.
      *
      * @param position Position of the new selected slide
      */
     protected void onPageSelected(int position) {
-        // Empty method
+        LogHelper.d(TAG, "Selected slide number: " + (position + 1));
+        canChangeLastSlide(position);
+    }
+
+
+    /**
+     * Called by {@link #onPageSelected(int)} to check if the user has requested permissions on the previous slide.
+     * This method is required as a safety check because otherwise,
+     * the user might skip a slide containing permissions.
+     * <p>
+     * This method will send the user back to the first slide that contains permissions.
+     *
+     * @param position - The position that the user has changed to.
+     *                 <p>
+     *                 TODO(1): This method will require tweaking when switching to Viewpager2 as
+     *                 there is no RTL support in the current viewpager and that messes up the position of slides.
+     *                 <p>
+     *                 If the position issue is resolved in Viewpager2, simply remove the code inside if(isRtl()).
+     */
+    private void canChangeLastSlide(int position) {
+        if (isRtl()) {
+            if (permissionsPositions.contains(position + 2) && !requestedSlides.contains(position + 2)) {
```

Comment:
The fact that you need to add a `+2` here should be documented. This would make the code harder to maintain and really error prone to off-by-one errors. 

Would be better if you could subtract -1 when you add the numbers with the `permissionsPositions.add` method. There would be clearer that the user will provide you a `1` but you expect to work with `0` when dealing with the first slide.

Reply:
Actually, I guess the RTL implementation is wrong, let me investigate

Reply:
@cortinico I am done, I can't wrap my head around this RTL thing in Viewpager. I am removing my RTL implementation as it's wrong. I think we should wait for Viewpager2 as it brings RTL support. So, in the meantime only ltr layouts have swipe support. There is no use implementing RTL if we have to remove it in a few weeks. Sorry, but I just can't do this.😭😭

Reply:
@cortinico RTL is fixed by the latest commit. Now it behaves normally. Please take a look😊😊